thrall/app/lib/elasticsearch/ElasticSearch.scala

```diff
@@ -37,15 +37,36 @@ class ElasticSearch(config: ElasticSearchConfig, metrics: Option[ThrallMetrics])
     executeAndLog(getAliases(Nil, Seq(alias)), s"Looking up index for alias '$alias'").map(_.result.mappings.keys.headOption)
   }
 
-  def bulkInsert(images: Seq[Image])(implicit ex: ExecutionContext, logMarker: LogMarker): List[Future[ElasticSearchBulkUpdateResponse]] = {
+  def setMigrationInfo(imageId: String, migrationInfo: Either[MigrationFailure, MigrationTo])(implicit ex: ExecutionContext, logMarker: LogMarker): Future[Response[Any]] = {
+    import lib.elasticsearch.EsInfo.eitherFormat
+
+    val migrationParam = Json.toJson(migrationInfo)
+    val scriptSource =
+      """
+        | if (ctx._source.esInfo != null && ctx._source.esInfo.migration != null) {
+        |   ctx._source.esInfo.remove("migration");
+        | }
+        | if (ctx._source.esInfo == null) {
+        |   ctx._source.esInfo = [:];
+        | }
+        | ctx._source.esInfo.migration = params.migration;
+        |""".stripMargin
+
+    val request = updateById(imagesCurrentAlias, imageId)
+      .script(Script(script = scriptSource, params = Map("migration" -> asNestedMap(migrationParam))).lang("painless"))
```

Comment:
now that we're capturing the index name associated with failures, this can be simplified as we no longer need to clear the `esInfo.migration` data before writing the new info

Reply:
@twrichards we've now simplified this and it's working locally.